### PR TITLE
Driver-only ecdsa wrapup

### PR DESCRIPTION
**Pull request by mpg**
## Description

This PR:
- fixes the issue found here: https://github.com/Mbed-TLS/mbedtls/pull/7117#discussion_r1131360315
- fixes a related issue that I noticed while looking at this
- adds a ChangeLog entry for the driver-only ECDSA work.

## Gatekeeper checklist

- [x] **changelog** provided - for the entire driver-only ECDSA series
- [x] **backport** not required - new feature
- [x] **tests** not required for `check_config.h` and ChangeLog

### CODE REVIEW

File: include/mbedtls/config_psa.h [hunk 1]
```diff
@@ -848,6 +843,11 @@ extern "C" {
 
 #endif /* MBEDTLS_PSA_CRYPTO_CONFIG */
 
+#if defined(PSA_WANT_ALG_ECDSA) && defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) && \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY)
+#define PSA_HAVE_FULL_ECDSA 1
+#endif
+
```
Comment (valeriosetti) [inline include/mbedtls/config_psa.h:850]: I'm OK with moving this code forward in the file, but I was wondering: is it possible to have `PSA_WANT` symbols when `MBEDTLS_PSA_CRYPTO_CONFIG` is not defined? 
I mean, without it we don't include `crypto_config.h` and also most of the `config_psa.h` is also skipped (at least the part which automatically sets `PSA_WANT` symbols).

**Review by valeriosetti — COMMENTED** [review]
I only have one generic question but the PR looks OK to me

File: include/mbedtls/config_psa.h (hunk 1 above)
Comment (gilles-peskine-arm) [reply include/mbedtls/config_psa.h:850]: We do have `PSA_WANT` whether `MBEDTLS_PSA_CRYPTO_CONFIG` is defined or not. The difference that `MBEDTLS_PSA_CRYPTO_CONFIG` makes is whether `PSA_WANT` symbols are calculated from `MBEDTLS` symbols or the other way round.

(I don't know how this applies specifically here.)

**Review by AndrzejKurek — APPROVED** [review]

**Review by valeriosetti — APPROVED** [review]

**Comment by mpg** [comment]
Open CI is fully green, so no need to wait for Internal to finish.

**Pull request merged**